bundle/src/main/java/com/adobe/acs/commons/cloudservices/pwa/impl/PwaServiceWorkerJavaScriptServlet.java

```diff
@@ -77,14 +75,26 @@ protected final void doGet(SlingHttpServletRequest request, SlingHttpServletResp
     }
 
     private void writeJavaScript(SlingHttpServletRequest request, SlingHttpServletResponse response) {
- 
-        final Configuration configuration = modelFactory.createModel(request, Configuration.class);
 
-        final Collection<ClientLibrary> htmlLibraries =
-                htmlLibraryManager.getLibraries(configuration.getServiceWorkerJsCategories(),
-                        LibraryType.JS, true, false);
+        final Configuration configuration = modelFactory.createModel(request, Configuration.class);
+        boolean serviceWorkerJsFlag = false;
+        Collection<ClientLibrary> htmlLibraries ;
+        for (String selector : request.getRequestPathInfo().getSelectors()) {
+            if ("service-worker".equals(selector)) {
+                serviceWorkerJsFlag = true;
 
-        if (htmlLibraries.size() > 0) {
+            }
+        }
+        if (serviceWorkerJsFlag) {
```

Comment:
Does this replace using the actual selected client library categories? What is the advantage of this?

this has the downsides of 1) more code to maintain 2) more code to test 3) more code to break 4) wont enjoy things like versioned clientlibs 5) its not clear what this servlet does (its not does 2 things, not 1)

Reply:
This change  was done to make the js that loads the SW pluggable.